MonoGame.Framework/Graphics/SpriteBatch.cs

```diff
@@ -253,23 +252,86 @@ void CheckValid(SpriteFont spriteFont, StringBuilder text)
 		{
             CheckValid(texture);
 
-            var w = texture.Width * scale.X;
-            var h = texture.Height * scale.Y;
-			if (sourceRectangle.HasValue)
-            {
-				w = sourceRectangle.Value.Width*scale.X;
-				h = sourceRectangle.Value.Height*scale.Y;
-			}
+            var item = _batcher.CreateBatchItem();
+            item.Texture = texture;
 
-            DrawInternal(texture,
-				new Vector4(position.X, position.Y, w, h),
-				sourceRectangle,
-				color,
-				rotation,
-				origin * scale,
-				effects,
-                layerDepth,
-				true);
+            // set SortKey based on SpriteSortMode.
+            switch ( _sortMode )
+            {
+                // Comparison of Texture objects.
+                case SpriteSortMode.Texture:
+                    item.SortKey = texture.SortingKey;
+                    break;
+                // Comparison of Depth
+                case SpriteSortMode.FrontToBack:
+                    item.SortKey = layerDepth;
+                    break;
+                // Comparison of Depth in reverse
+                case SpriteSortMode.BackToFront:
+                    item.SortKey = -layerDepth;
+                    break;
+            }
+                        
+            origin = origin * scale;
+            
+            float w, h;
+            if (sourceRectangle.HasValue)
+            {
+				var srcRect = sourceRectangle.Value;
```

Comment:
Since this is all about micro-optimisations, I thought I'd mention that `GetValueOrDefault()` is actually slightly faster than `Value` - https://ericlippert.com/2012/12/20/nullable-micro-optimizations-part-one/